client-ts/signalr/src/HttpConnection.ts

```diff
@@ -160,8 +160,12 @@ export class HttpConnection implements IConnection {
             const transferFormats = endpoint.transferFormats.map((s) => TransferFormat[s]);
             if (!requestedTransport || transport === requestedTransport) {
                 if (transferFormats.indexOf(requestedTransferFormat) >= 0) {
-                    this.logger.log(LogLevel.Trace, `Selecting transport '${TransportType[transport]}'`);
-                    return transport;
+                    if (transport === TransportType.WebSockets && typeof WebSocket === "undefined") {
```

Comment:
Need the same check for `transport === TransportType.ServerSentEvents && typeof EventSource === "undefined"`, but only in TypeScript (because we have to use the `EventSource` type to implement it, we can't use `XMLHTTPRequest` to implement SSE)